src/finance/PoolSteward.sol

```diff
@@ -0,0 +1,191 @@
+
+    /// @inheritdoc IPoolSteward
+    ICollector public immutable COLLECTOR;
+
+    /// @inheritdoc IPoolSteward
+    mapping(address poolV3 => bool) public poolsV3;
+    mapping(address poolV2 => bool) public poolsV2;
+
+    constructor(
+        address _owner,
+        address _guardian,
+        address collector,
+        address[] memory _poolsV2,
+        address[] memory _poolsV3
+    ) {
+        _transferOwnership(_owner);
+        _updateGuardian(_guardian);
+
+        COLLECTOR = ICollector(collector);
+
+        for (uint256 i; i < _poolsV2.length; i++) {
+            _approvePool(_poolsV2[i], false);
+        }
+
+        for (uint256 i; i < _poolsV3.length; i++) {
+            _approvePool(_poolsV3[i], true);
+        }
+    }
+
+    /// Steward Actions
+
+    /// @inheritdoc IPoolSteward
+    function depositV3(
+        address pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+        _validateV3Pool(pool);
+
+        CU.IOInput memory depositData = CU.IOInput(pool, reserve, amount);
+        CU.depositToV3(COLLECTOR, depositData);
+    }
+
+    /// @inheritdoc IPoolSteward
+    function withdrawV3(
+        address pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        CU.IOInput memory withdrawData = CU.IOInput(pool, reserve, amount);
+
+        CU.withdrawFromV3(COLLECTOR, withdrawData, address(COLLECTOR));
+    }
+
+    /// @inheritdoc IPoolSteward
+    function withdrawV2(
+        address pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV2Pool(pool);
+
+        CU.IOInput memory withdrawData = CU.IOInput(pool, reserve, amount);
+        CU.withdrawFromV2(COLLECTOR, withdrawData, address(COLLECTOR));
+    }
+
+    /// @inheritdoc IPoolSteward
+    function migrateBetweenV3(
+        address fromPool,
+        address toPool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV3Pool(toPool);
+
+        CU.IOInput memory withdrawData = CU.IOInput(fromPool, reserve, amount);
+        uint256 withdrawnAmt = CU.withdrawFromV3(
+            COLLECTOR,
+            withdrawData,
+            address(COLLECTOR)
+        );
+
+        CU.IOInput memory depositData = CU.IOInput(
+            toPool,
+            reserve,
+            withdrawnAmt
+        );
+        CU.depositToV3(COLLECTOR, depositData);
+    }
+
+    /// @inheritdoc IPoolSteward
+    function migrateV2toV3(
+        address v2Pool,
+        address v3Pool,
+        address reserve,
+        uint256 amount
+    ) external onlyOwnerOrGuardian {
+        if (amount == 0) revert InvalidZeroAmount();
+
+        _validateV2Pool(v2Pool);
```

Comment:
why validate the source pool here, but not in `migrateBetweenV3` - i'm not 100% sure if is needed or not.